src/back-end/lib/resources/affiliation.ts

```diff
@@ -131,40 +124,38 @@ const resource: Resource = {
 
   update(connection) {
     return {
-      async parseRequestBody(request) {
+      async parseRequestBody(request): Promise<null> {
         return null;
       },
-      async validateRequestBody(request) {
-        const validatedAffiliationId = await validateAffiliationId(connection, request.params.id);
-        if (validatedAffiliationId.tag === 'invalid') {
+      async validateRequestBody(request): Promise<Validation<ValidatedUpdateRequestBody, UpdateValidationErrors>> {
```

Comment:
You shouldn't need to type the return value of this function, as it should be inferred by the `Resource` type above.